contracts/pegged/OriginalTokenVaults.sol

```diff
@@ -7,21 +7,32 @@ import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
 import "@openzeppelin/contracts/security/ReentrancyGuard.sol";
 import "../interfaces/ISigsVerifier.sol";
 import "../libraries/PbPegged.sol";
+import "../safeguard/Pauser.sol";
+import "../safeguard/VolumeControl.sol";
+import "../safeguard/DelayedTransfer.sol";
 
 /**
  * @title the vaults to lock original tokens at the source chain
  */
-contract OriginalTokenVaults is ReentrancyGuard {
+contract OriginalTokenVaults is ReentrancyGuard, Pauser, VolumeControl, DelayedTransfer {
     using SafeERC20 for IERC20;
 
     ISigsVerifier public immutable sigsVerifier;
 
-    uint64 public mintseq; // ensure unique Mint event, start from 1
+    mapping(bytes32 => bool) public records;
+
+    // TODO: may remove valuts as records are kept in sgn
```

Comment:
Does this still make sense now that `ref_chain_id` has multiple meanings?

Reply:
removed